lib/device_detector/device.rb

```diff
@@ -0,0 +1,54 @@
+class DeviceDetector
+  class Device < Parser
+
+    def known?
+      regex_meta.any?
+    end
+
+    def name
+      ModelExtractor.new(user_agent, regex_meta).call
+    end
+
+    def device_type
+      regex_meta['device']
+    end
+
+    private
+
+    def filenames
+      [
+        'devices/mobiles.yml'
+      ]
+    end
+
+    # Overwrite the default parser handling logic
+    def self.regexes_for(filepaths)
+      @regexes ||=
+        begin
+          regexes = YAML.load(filepaths.map { |filepath| File.read(filepath) }.join)
+
+          recursive_regex_parser(regexes)
```

Comment:
I have an idea on how to avoid overriding the whole `.regexes_for(:filepaths)` method.

In `lib/device_detector/parser.rb` we could rewrite the method like this:

```ruby
    def self.regexes_for(filepaths)
      @regexes ||=
        begin
          regexes = YAML.load(filepaths.map { |filepath| File.read(filepath) }.join)
          parse_regexes(regexes)
        end
    end

    def self.parse_regexes(regexes)
        regexes.map do |meta|
          meta['regex'] = Regexp.new(meta['regex'])
          meta
        end
    end
```

This would allow to just override the `.parse_regexes(regexes)` method.